plugins/azure-devops-backend/src/api/AzureDevOpsApi.ts

```diff
@@ -49,23 +49,70 @@ import {
 import { TeamMember as AdoTeamMember } from 'azure-devops-node-api/interfaces/common/VSSInterfaces';
 import { Logger } from 'winston';
 import { PolicyEvaluationRecord } from 'azure-devops-node-api/interfaces/PolicyInterfaces';
-import { WebApi } from 'azure-devops-node-api';
+import {
+  WebApi,
+  getBearerHandler,
+  getPersonalAccessTokenHandler,
+} from 'azure-devops-node-api';
 import {
   TeamProjectReference,
   WebApiTeam,
 } from 'azure-devops-node-api/interfaces/CoreInterfaces';
 import { UrlReader } from '@backstage/backend-common';
+import { Config } from '@backstage/config';
+import {
+  DefaultAzureDevOpsCredentialsProvider,
+  ScmIntegrations,
+} from '@backstage/integration';
 
 /** @public */
 export class AzureDevOpsApi {
-  public constructor(
-    private readonly logger: Logger,
-    private readonly webApi: WebApi,
-    private readonly urlReader: UrlReader,
-  ) {}
+  private readonly logger: Logger;
+  private readonly urlReader: UrlReader;
+  private readonly config: Config;
+  private readonly credentialsProvider: DefaultAzureDevOpsCredentialsProvider;
+
+  private constructor(logger: Logger, urlReader: UrlReader, config: Config) {
+    this.logger = logger;
+    this.urlReader = urlReader;
+    this.config = config;
+
+    const scmIntegrations = ScmIntegrations.fromConfig(this.config);
+    this.credentialsProvider =
+      DefaultAzureDevOpsCredentialsProvider.fromIntegrations(scmIntegrations);
```

Comment:
these are good to do in fromConfig, since that leaves the constructor with just basic assignments, and also lets the constructor more easily receive higher-level mocks in testing

Reply:
speaking of which, don't the actual api calls of this client ever get exercised by tests? 🤔

Reply:
I've moved these to `fromConfig`. As for tests, I didn't have any when I initially contributed the plugin. I'll work on adding some 👍

Reply:
Tests have been added 🚀